File: remove_logo_bg.js
const sharp = require('sharp');

async function removeBackground() {
  const inputPath = 'shinigamilogo.png';
  const outputPath = 'public/logo.png';

  const { data, info } = await sharp(inputPath)
    .ensureAlpha()
    .raw()
    .toBuffer({ resolveWithObject: true });

  const { width, height, channels } = info;
  const pixels = new Uint8Array(data);

  // Sample the top-left corner to detect the exact background color
  const bgR = pixels[0], bgG = pixels[1], bgB = pixels[2];
  console.log(`Detected background color: rgb(${bgR}, ${bgG}, ${bgB})`);

  for (let i = 0; i < width * height; i++) {
    const offset = i * channels;
    const r = pixels[offset];
    const g = pixels[offset + 1];
    const b = pixels[offset + 2];

    // Distance from detected background color
    const dist = Math.sqrt(
      Math.pow(r - bgR, 2) +
      Math.pow(g - bgG, 2) +
      Math.pow(b - bgB, 2)
    );

    // Threshold: pixels close to bg color become transparent
    // Tolerance of 80 covers navy gradient + dark shadow variations
    const tolerance = 80;
    if (dist < tolerance) {
      // Soft edge: fully transparent at center, fade out at edge
      const alpha = Math.round(Math.min(1, (dist / tolerance) * 3.0) * 255);
      pixels[offset + 3] = alpha;
    }

    // Second pass: also zap any leftover dark/gray noise pixels
    // that have very low saturation and low brightness
    const lum = 0.299 * r + 0.587 * g + 0.114 * b;
    const maxC = Math.max(r, g, b);
    const minC = Math.min(r, g, b);
    const saturation = maxC === 0 ? 0 : (maxC - minC) / maxC;
    if (lum < 50 && saturation < 0.35) {
      // Dark desaturated pixel = residue, make transparent
      const alpha = Math.round(Math.min(1, lum / 50) * 255);
      pixels[offset + 3] = Math.min(pixels[offset + 3], alpha);
    }
  }

  await sharp(pixels, { raw: { width, height, channels } })
    .png()
    .toFile(outputPath);

  console.log(`✅ Transparent logo saved to: ${outputPath}`);
}

removeBackground().catch(console.error);

File: remove_logo_bg.py
"""
Remove the dark navy background from the Shinigami Chess logo
and make it transparent so it blends with any website background.
"""

from PIL import Image
import numpy as np

input_path = "shinigamilogo.png"
output_path = "public/logo.png"

img = Image.open(input_path).convert("RGBA")
data = np.array(img, dtype=np.float32)

R, G, B, A = data[:,:,0], data[:,:,1], data[:,:,2], data[:,:,3]

# The logo background is dark navy (~10-20, 18-30, 40-60 in RGB)
# The gold artwork is brighter and warmer
# We remove pixels that are dark AND blue-dominant

# Brightness (luminance)
lum = 0.299*R + 0.587*G + 0.114*B

# "Blue-dominant dark" = typical navy background
is_dark = lum < 60
is_bluish = B > R * 1.3  # blue channel dominates red
is_bg = is_dark & is_bluish

# Soft transition: pixels near the edge of the mask get partial alpha
# Compute how "background-like" each pixel is (0=artwork, 1=bg)
bg_score = (np.clip(60 - lum, 0, 60) / 60.0) * (np.clip(B - R * 1.3, 0, 30) / 30.0)

# Alpha: artwork pixels stay fully opaque, bg pixels become fully transparent
alpha_new = np.clip(1.0 - bg_score * 2.5, 0.0, 1.0) * 255

data[:,:,3] = alpha_new.astype(np.uint8)

result = Image.fromarray(data.astype(np.uint8), "RGBA")
result.save(output_path, "PNG")
print(f"Saved transparent logo to: {output_path}")
